x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/idp/SamlIdentityProvider.java

```diff
@@ -7,10 +7,18 @@
 package org.elasticsearch.xpack.idp.saml.idp;
 
 
+import org.opensaml.security.x509.X509Credential;
+
 /**
  * SAML 2.0 configuration information about this IdP
  */
 public interface SamlIdentityProvider {
 
     String getEntityId();
+
+    String getSingleSignOnEndpoint(String binding);
```

Comment:
Could these return `URI` instead?

Also where do we expect to need these? Is it just for metadata? (which is a real need - I just want to check how we expect to consume them).

Reply:
see https://github.com/elastic/elasticsearch/pull/51682#discussion_r373320101

> Also where do we expect to need these? Is it just for metadata? 

Metadata creation and incoming SAML message validation